app/config/config_test.yml

```diff
@@ -1,6 +1,7 @@
 parameters:
     AdapterSecurityAdminClass:                LegacyTests\PrestaShopBundle\Mock\AdapterSecurityAdminMock
     prestashop.security.voter.product.class:  LegacyTests\PrestaShopBundle\Mock\PageVoter
+    ps_tmp_img_dir: "/tmp/"
```

Comment:
Nope :scream:

Reply:
Did I misunderstood @matks here ?

We will probably need to remove hardcoded _PS_TMP_IMG_DIR_ constants and inject them through the constructor. This will allow us to modify the working directory used by CoverImageUploader and replace it by a virtual one provided by vFsStream.

We will be able to inject our virtual directory by 3 possible ways:

either override Symfony service declaration for this service, only in test environment
control how _PS_TMP_IMG_DIR_ is injected into the service, for example if we use a Symfony parameter placeholder like `%prestashop.temporary.image.diectoryr% we could modify the value of this placeholder for test environment

Reply:
Talked with Rokas and yes we agree it is a bad idea to use "/tmp/". Actually could use relative /img/tmp/ path from root and while vfstream does not support tempnam() function which is used by our legacy code it seems there's no benefit of injecting different urls for test environment then. Do we understand it correctly? @PierreRambaud @matks would be nice to confirm the solution now because it is already clear what kind of problems we face.

Reply:
Actually it has some meaning injecting parameters via services. While injecting %prestashop.temporary.image.diectoryr% is not so important because it's already temporary. 

Also in order to get the EditableCategory GetCategoryForEditing there is one more constant _PS_CAT_IMG_DIR_ which is in the /img/c/ folder and injecting the parameter here is useful for test to be able to confirm image was uploaded or already deleted

There is one more nice to have I am searching for - easy way to get these parameter in tests. Like symfony controller has $this->getParameter() method. Looking for similar way in tests because to inject a service for just getting the parameter might seem too much.

Reply:
Always use the `var/` for temporary files, the  `img` directory is not for temporary resources./
Maybe time to have a `var/tmp`